src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -375,6 +376,21 @@ static void read_channel_args(grpc_chttp2_transport* t,
                 .GetObjectRef<grpc_core::channelz::SocketNode::Security>());
   }
 
+  const int soft_limit =
+      channel_args.GetInt(GRPC_ARG_MAX_METADATA_SIZE).value_or(-1);
+  if (soft_limit == -1) {
```

Comment:
maybe make this `<=0`

Reply:
Hm I'm worried about consistency then - if the value is 0, that means the user set the soft limit to that. I think we should treat any user-set value the same way; otherwise, they would see different behavior for a 0 vs. some other value. The user can also set the hard limit to 0 right now, so whatever we decide, we should do for both.